### PR TITLE
Delete org when error occurs running flow

### PR DESCRIPTION
**Pull request by Br4nd0R**
# Issues Closed
* CumulusCI will now remove orgs when the `--delete-org` option is passed to `cci flow run`, even if an error occurs while running the flow.

## Dev Notes
I may be missing something here so let me know if I'm way off base. Not sure if there's a good reason we had just a `try/finally` block without an `except` clause in there. It looks like we would always display the "flow complete" message even when a flow encountered an error.

### CODE REVIEW

**Review by davisagli — CHANGES_REQUESTED** [review]
@Br4nd0R The fix I had in mind for this was to indent the "if delete_org:" block so that it is inside the "finally:" block and will run even if the flow raised an exception.

I don't think we want to suppress the raising of the exception. That will prevent the stacktrace from being captured in the cci log.

And I don't think we need to explicitly log the exception here. That is already done within the flowrunner.

**Review by davisagli — APPROVED** [review]
LGTM.

If you also want to move `runtime.alert(f"Flow Complete: {flow_name}")` outside of `finally` so that we only log that when the flow completed without an exception, I'm okay with that too.

File: cumulusci/cli/flow.py [hunk 1]
```diff
@@ -170,3 +168,5 @@ def flow_run(runtime, flow_name, org, delete_org, debug, o, skip, no_prompt):
                     "Scratch org deletion failed.  Ignoring the error below to complete the flow:"
                 )
                 click.echo(str(e))
+
+    runtime.alert(f"Flow Complete: {flow_name}")
```
Comment (davisagli) [inline cumulusci/cli/flow.py:172]: @Br4nd0R sorry for the back and forth on this but I just realized we also should call `runtime.alert` with a different message when there is an error. It's the thing that causes an OS notification to pop up and we wouldn't want to lose that when there's an error.

Comment (Br4nd0R) [reply cumulusci/cli/flow.py:172]: no problem

**Review by davisagli — APPROVED** [review]

**Pull request merged**